ddpui/celeryworkers/tasks.py

```diff
@@ -871,10 +873,175 @@ def summarize_logs(
+                "results": {},
+            }
+        )
+        return
+
+    orguser = OrgUser.objects.filter(id=orguser_id).first()
+    org = orguser.org
+
+    # create a partial session
+    llm_session = LlmSession.objects.create(
+        request_uuid=self.request.id,
+        orguser=orguser,
+        org=org,
+        session_status=LlmSessionStatus.RUNNING,
+        session_type=LlmAssistantType.LONG_TEXT_SUMMARIZATION,
+        request_meta={"sql": sql},
+    )
+
+    credentials = secretsmanager.retrieve_warehouse_credentials(org_warehouse)
+
+    try:
+        wclient = WarehouseFactory.connect(credentials, wtype=org_warehouse.wtype)
+    except Exception as err:
+        logger.error("Failed to connect to the warehouse - %s", err)
+        taskprogress.add(
+            {
+                "message": "Failed to connect to the warehouse",
+                "status": TaskProgressStatus.FAILED,
+                "result": None,
+            }
+        )
+        return
+
+    # fetch the results of the query
+    logger.info(f"Submitting query to warehouse for execution \n '''{sql}'''")
+    rows = []
+    try:
+        rows = wclient.execute(sql)
+    except Exception as err:
+        logger.error(err)
+        taskprogress.add(
+            {
+                "message": str(err),
+                "status": TaskProgressStatus.FAILED,
+                "result": None,
+            }
+        )
+        llm_session.session_status = LlmSessionStatus.FAILED
+        llm_session.save()
+        return
+
+    if len(rows) == 0:
+        taskprogress.add(
+            {
+                "message": "No results found for the query",
+                "status": TaskProgressStatus.FAILED,
+                "result": None,
+            }
+        )
+        llm_session.session_status = LlmSessionStatus.FAILED
+        llm_session.save()
+        return
+
+    try:
+        # prepare the text data to be uploaded
+        headers = rows[0].keys()
+        csv_lines = [",".join(headers) + "\n"]  # Add header row
+        for item in rows:
+            row = [
+                str(item[header]) for header in headers
+            ]  # Extract values in header order
+            csv_lines.append(" ".join(row) + "\n")
+
+        # upload the results as a file to llm service
+        fpath, session_id = llm_service.upload_text_as_file(
+            "".join(csv_lines), "warehouse_results"
+        )
+        logger.info("Uploaded file successfully to LLM service at " + str(fpath))
+        logger.info("Session ID: " + session_id)
+
+        # query the llm service with two prompts
+        user_prompts = [user_prompt]
+
+        assistant_prompt = AssistantPrompt.objects.filter(
+            type=LlmAssistantType.LONG_TEXT_SUMMARIZATION
+        ).first()
+        if not assistant_prompt:
+            raise Exception(
+                "Assistant/System prompt not found for warehouse summarization"
+            )
+
```

Comment:
in case the llm service conks out should we store the `user_prompts` and `assistant_prompt` here instead of below

Reply:
and the `session_id`